WalletWasabi.Fluent/Extensions/PrivacyModeHelper.cs

```diff
@@ -0,0 +1,51 @@
+using System.Reactive.Linq;
+
+namespace WalletWasabi.Fluent.Extensions;
+
+public static class PrivacyModeHelper
+{
+	private static readonly TimeSpan RevealDelay = TimeSpan.FromSeconds(0.75);
+	private static readonly TimeSpan HideDelay = TimeSpan.FromSeconds(10);
+
+	public static IObservable<bool> DelayedRevealAndHide(
+		IObservable<bool> pointerOver,
+		IObservable<bool> isPrivacyModeEnabled,
+		IObservable<bool>? isForced = null)
+	{
+		isForced ??= Observable.Return(false);
+		var isPointerOver = pointerOver
+			.Select(
+				isTrue => isTrue
+					? PointerOverObs()
+					: PointerOutObs())
+			.Switch();
+
+		var displayContent = isPrivacyModeEnabled
+			.CombineLatest(
+				isPointerOver,
+				isForced,
+				(privacyModeEnabled, pointerOver, forced) => !privacyModeEnabled || pointerOver || forced);
+
+		return displayContent;
+	}
+
+	private static IObservable<bool> PointerOutObs()
+	{
+		return Observable.Return(false);
+	}
+
+	private static IObservable<bool> PointerOverObs()
+	{
+		var hideObs =
+			Observable
+				.Return(false)
+				.Delay(HideDelay);
+
+		var showObs =
+			Observable
+				.Return(true)
+				.Delay(RevealDelay);
+
+		return showObs.Concat(hideObs);
+	}
+}
```

Comment:
what's the reasoning behind this complex combination of observables? Seems to me the previous code was much more straightforward:

```csharp
Services.UiConfig
			.WhenAnyValue(x => x.PrivacyMode)
			.Merge(this.WhenAnyValue(x => x.ForceShow))
			.ObserveOn(RxApp.MainThreadScheduler)
			.Subscribe(_ =>
			{
				var value = !Services.UiConfig.PrivacyMode || ForceShow;

				IsPrivacyContentVisible = !value;
				IsContentVisible = value;
			})
			.DisposeWith(_disposable);
```

Reply:
The previous code doesn't deal with time at all. The new code is more complex because 2 different delays have to be introduced to get the functionality.

In any case, I will try to improve the readability. The automatic formatting has inserted some ugly indentations...

NOTE: The previous code has some logic into the Subscribe method. That's a bad practice.